packages/in_app_purchase/example/lib/main.dart

```diff
@@ -361,28 +365,32 @@ class _MyAppState extends State<MyApp> {
 
   void _listenToPurchaseUpdated(List<PurchaseDetails> purchaseDetailsList) {
     purchaseDetailsList.forEach((PurchaseDetails purchaseDetails) async {
+      await InAppPurchaseConnection.instance.consumePurchase(purchaseDetails);
       if (purchaseDetails.status == PurchaseStatus.pending) {
         showPendingUI();
       } else {
         if (purchaseDetails.status == PurchaseStatus.error) {
           handleError(purchaseDetails.error);
+          return;
```

Comment:
Also on iOS completePurchase() needs to be called for status == PurchaseStatus.error. With this change this will be broken. 

Please refer to completePurchase documentation. 
https://github.com/flutter/plugins/blob/ed7620cc51f51128e44d5eef42f590c8d76c5c64/packages/in_app_purchase/lib/src/in_app_purchase/in_app_purchase_connection.dart#L196